src/dodal/devices/undulator.py

```diff
@@ -12,17 +14,28 @@ class UndulatorGapAccess(Enum):
     DISABLED = "DISABLED"
 
 
-class Undulator(Device):
-    gap_motor = Component(EpicsMotor, "BLGAPMTR")
-    current_gap = Component(EpicsSignalRO, "CURRGAPD")
-    gap_access = Component(EpicsSignalRO, "IDBLENA")
-    gap_discrepancy_tolerance_mm: float = UNDULATOR_DISCREPANCY_THRESHOLD_MM
+class Undulator(StandardReadable):
+    """
+    An Undulator-type insertion device, used to control photon emission at a given
+    beam energy.
+    """
 
     def __init__(
         self,
+        prefix: str,
+        name: str = "",
         lookup_table_path="/dls_sw/i03/software/daq_configuration/lookup/BeamLine_Undulator_toGap.txt",
-        *args,
-        **kwargs,
-    ):
-        super().__init__(*args, **kwargs)
+    ) -> None:
+        self.gap_motor = Motor(prefix + "BLGAPMTR")
+        self.current_gap = epics_signal_r(float, prefix + "CURRGAPD")
+        self.gap_access = epics_signal_r(UndulatorGapAccess, prefix + "IDBLENA")
```

Comment:
Must: The `Enum` needs to inherit from `str` too. +1 to https://github.com/bluesky/ophyd-async/issues/261